crates/symbolicator/src/endpoints/applecrashreport.rs

```diff
@@ -46,8 +46,11 @@ async fn handle_apple_crash_report_request(
     let report = report.ok_or_else(|| error::ErrorBadRequest("missing apple crash report"))?;
 
     let symbolication = state.symbolication();
-    let request_id =
-        symbolication.process_apple_crash_report(params.scope, report, sources, options);
+    let request_id = symbolication
+        .process_apple_crash_report(params.scope, report, sources, options)
+        .ok_or_else(|| {
+            error::ErrorServiceUnavailable("maximum number of concurrent requests reached")
+        })?;
```

Comment:
I’m wondering why we `ok_or_else` here, and why not just straight up return the error from the appropriate functions.